### PR TITLE
[boschshc] Add support for Smoke Detector II

### PR DESCRIPTION
**Pull request by pat-git023**
# Add Smoke Detector 2

# Description
This PR adds support of the smoke detector 2 to the Bosch Smart Home Binding (#16243 )

# Testing

Your pull request will automatically be built and available under the following folder:
[org.openhab.binding.boschshc/4.2.0-SNAPSHOT](https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.boschshc/4.2.0-SNAPSHOT/)

### CODE REVIEW

**Comment by pat-git023** [comment]
@david-pace 
During testing I noticed a little difference for the communication quality level. Even though my signal strength is 3 the item in openHAB shows it as '_good_'. From the state enum `org.openhab.binding.boschshc.internal.services.communicationquality.dto.CommunicationQualityState` I would have excpeted the state to be '_NORMAL_'.  
Do you know where the mapping for the UI happens?
Screenshot 2024-02-03 at 08 53 48
Screenshot 2024-02-03 at 08 54 08

**Comment by jlaur** [comment]
> Even though my signal strength is 3 the item in openHAB shows it as 'good'.

Perhaps this can help with some context: https://github.com/openhab/openhab-addons/pull/16093#discussion_r1434107400

**Comment by david-pace** [comment]
Interestingly the exact same question was asked in https://github.com/openhab/openhab-addons/issues/14562#issuecomment-1924125575. We use an openHAB system channel and translate the Bosch-specific communication quality to an openHAB-specific signal strength as discussed [here](https://github.com/openhab/openhab-addons/pull/16093#discussion_r1434107400).

The mapping is:

    UNKNOWN -> 0
    BAD -> 1
    MEDIUM -> 2
    NORMAL -> 3
    GOOD -> 4

As you can see, GOOD is the best Bosch communication quality, so it is translated to the best openHAB signal strength 4 which is "excellent". In my opinion this is the best mapping but if you have a better idea feel free to propose a different mapping.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smokedetector/SmokeDetector2Handler.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    protected void initializeServices() throws BoschSHCException {
+        super.initializeServices();
+
+        this.createService(CommunicationQualityService::new, this::updateChannels, List.of(CHANNEL_SIGNAL_STRENGTH),
+                true);
+    }
+
+    private void updateChannels(CommunicationQualityServiceState communicationQualityServiceState) {
+        logger.debug("***** update communication quality with {}", communicationQualityServiceState.quality);
```
Comment (david-pace) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smokedetector/SmokeDetector2Handler.java:51]: Is this log really needed?
If so, maybe we should remove the asterisks and Capitalize all log messages: `Updating communication quality for Smoke Detector II with value {}`

**Review by david-pace — APPROVED** [review]
Looks great 😎
Just some things to double-check:
* did you re-generate the i18n file in the end after having edited everything in `thing-types.xml`?
* Did you check that there are no SAT warnings in `target/code-analysis/report.html`?

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smokedetector/SmokeDetector2Handler.java (hunk 1 above)
Comment (pat-git023) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smokedetector/SmokeDetector2Handler.java:51]: upps, that was just a debug while I did some troubleshooting and used the asterisks to find it faster in the log 😬 

will remove the whole log

Comment (pat-git023) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smokedetector/SmokeDetector2Handler.java]: log has been removed

**Comment by pat-git023** [comment]
- i18n was regenerated
- no findings in the report: https://ci.openhab.org/job/PR-openHAB-Addons/19505/Static_20Code_20Analysis_20Report/

**Review by jlaur — APPROVED** [review]
Thanks for adding support for this new device!

**Pull request merged**